sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -148,6 +159,25 @@ object TypeCoercion {
     case (l, r) => None
   }
 
+  val findCommonTypeToCompatibleWithHive: (DataType, DataType) => Option[DataType] = {
+    // Follow hive's binary comparison action:
+    // https://github.com/apache/hive/blob/rel/storage-release-2.4.0/ql/src/java/
+    // org/apache/hadoop/hive/ql/exec/FunctionRegistry.java#L781
```

Comment:
I saw the change history of this file. It sounds like Hive's type coercion rules are also evolving.